### PR TITLE
.travis.yml:complete refactor

### PR DESCRIPTION
**Pull request by andy5995**
When the repo is added, the public key isn't imported automatically. This removes the need to use the "allow-unauthenticated" option.

### CODE REVIEW

**Comment by andy5995** [comment]
Added gcc 8 because Slackware current and Debian Buster use it.

**Comment by andy5995** [comment]
hmm... testing something.....

**Comment by andy5995** [comment]
> hmm... testing something.....

How about another test.

**Comment by andy5995** [comment]
> > hmm... testing something.....
> 
> How about another test.

All right then

File: mk/linux/setupBuildDeps.sh [hunk 1]
```diff
@@ -145,7 +145,7 @@ case $distribution in
 				;;
 			14.04*)
 				#LTS, name > trusty, EoL April 2019
-				installcommand="apt-get install $APT_OPTIONS build-essential cmake libsdl2-dev libalut-dev libgl1-mesa-dev libglu1-mesa-dev libvorbis-dev libwxgtk3.0-dev libx11-dev liblua5.1-0-dev libjpeg-dev libpng12-dev libcurl4-gnutls-dev libxml2-dev libircclient-dev libglew-dev libftgl-dev libfribidi-dev libvlc-dev libvlccore-dev libcppunit-dev"
+				installcommand="apt-get install $APT_OPTIONS build-essential cmake libalut-dev libgl1-mesa-dev libglu1-mesa-dev libvorbis-dev libwxgtk3.0-dev libx11-dev liblua5.1-0-dev libjpeg-dev libpng12-dev libcurl4-gnutls-dev libxml2-dev libircclient-dev libglew-dev libftgl-dev libfribidi-dev libvlc-dev libvlccore-dev libcppunit-dev"
```
Comment (andy5995) [inline mk/linux/setupBuildDeps.sh:148]: Are we still supporting trusty? I had to remove the libsdl2-dev dep to get the travis build to work. the dpendency won't install on trusty due to conflicts, so the source tarball is downloaded and built instead.

If we're still supporting trusty, we may need to add that to this script the way it's done in .travis.yml in the `before_install` section.

File: .travis.yml [hunk 2]
```diff
@@ -1,85 +1,115 @@
...
-
-sudo: required
-
+cache: ccache
 matrix:
   include:
     - os: linux
+      dist: trusty
+      compiler: gcc
+      env: DEFAULT_TRAVIS_COMPILER="true" TRAVIS_DIST="trusty"
```
Comment (andy5995) [inline .travis.yml]: The default uses gcc-4. Do we want to keep a test for that or just test gcc-5-8?

**Comment by andy5995** [comment]
This is passing and ready for review. I left some line comments though with some additional questions that may need to be answered before merging.

**Comment by andy5995** [comment]
Reminder to whoever merges this.. use SQUASH & merge! ;)

File: mk/linux/setupBuildDeps.sh (hunk 1 above)
Comment (tomreyn) [reply mk/linux/setupBuildDeps.sh:148]: I think we only ever meant to support OS releases during their lifespans.

Travis CI build nodes used to be running old, and default to outdated, Ubuntu releases, though. Fortunately this [does not seem to be the case now](https://docs.travis-ci.com/user/reference/overview/#virtualization-environments) (defaults to Ubuntu 16.04 LTS "Xenial", Ubuntu 18.04 LTS "Bionic" is available).

File: .travis.yml (hunk 2 above)
Comment (tomreyn) [reply .travis.yml]: Reviewing GCC default versions currently in *supported* [Debian](https://qa.debian.org/madison.php?package=gcc&table=debian) and [Ubuntu](https://qa.debian.org/madison.php?package=gcc&table=ubuntu) releases, I think the oldest gcc version MegaGlest should currently try to support is Ubuntu 16.04 (LTS) "Xenial"'s **gcc 5.3.1** (until 04/2021, when Xenial reaches EOL).

**Comment by Jammyjamjamman** [comment]
Hey @andy5995, would it be possible to add ubuntu 18.04 and 20.04 to the tests? I think this should be good to merge, then. (It could be merged I think anyway, tbh.)

**Comment by andy5995** [comment]
> Hey @andy5995, would it be possible to add ubuntu 18.04 and 20.04 to the tests? I think this should be good to merge, then. (It could be merged I think anyway, tbh.)

Hi @Jammyjamjamman ;) Thanks for reviewing!

I rebased and added a test for g++ 10.

The test for 7, 9, and 10 will use bionic (18.04). The Ubuntu Linux build environment on Travis doesn't yet have 20.04 available for testing. https://docs.travis-ci.com/user/reference/linux/

**Comment by andy5995** [comment]
This is obsolete now that Github Actions is being used.

**Pull request closed without merging**